packages/flutter/lib/src/material/bottom_sheet.dart

```diff
@@ -677,37 +680,75 @@ class ModalBottomSheetRoute<T> extends PopupRoute<T> {
   }
 
   @override
-  Widget buildPage(BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation) {
+  Widget buildPage(BuildContext context, Animation<double> animation,
+      Animation<double> secondaryAnimation) {
+    final GlobalKey bottomSheetKey = GlobalKey();
+    final MaterialLocalizations localizations = MaterialLocalizations.of(context);
     final Widget content = DisplayFeatureSubScreen(
       anchorPoint: anchorPoint,
-      child: Builder(
-        builder: (BuildContext context) {
-          final BottomSheetThemeData sheetTheme = Theme.of(context).bottomSheetTheme;
-          final BottomSheetThemeData defaults = Theme.of(context).useMaterial3 ? _BottomSheetDefaultsM3(context) : const BottomSheetThemeData();
-          return _ModalBottomSheet<T>(
-            route: this,
-            backgroundColor: backgroundColor ?? sheetTheme.modalBackgroundColor ?? sheetTheme.backgroundColor ?? defaults.backgroundColor,
-            elevation: elevation ?? sheetTheme.modalElevation ?? defaults.modalElevation ?? sheetTheme.elevation,
-            shape: shape,
-            clipBehavior: clipBehavior,
-            constraints: constraints,
-            isScrollControlled: isScrollControlled,
-            enableDrag: enableDrag,
-          );
-        },
+      child: Stack(
+        children: <Widget>[
+          Builder(
+            builder: (BuildContext context) {
+              final BottomSheetThemeData sheetTheme =
+                  Theme.of(context).bottomSheetTheme;
+              final BottomSheetThemeData defaults =
+                  Theme.of(context).useMaterial3
+                      ? _BottomSheetDefaultsM3(context)
+                      : const BottomSheetThemeData();
+              return _ModalBottomSheet<T>(
+                route: this,
+                backgroundColor: backgroundColor ??
+                    sheetTheme.modalBackgroundColor ??
+                    sheetTheme.backgroundColor ??
+                    defaults.backgroundColor,
+                elevation: elevation ??
+                    sheetTheme.modalElevation ??
+                    defaults.modalElevation ??
+                    sheetTheme.elevation,
+                shape: shape,
+                clipBehavior: clipBehavior,
+                constraints: constraints,
+                isScrollControlled: isScrollControlled,
+                enableDrag: enableDrag,
+                bottomSheetKey: bottomSheetKey,
+              );
+            },
+          ),
+          LayoutBuilder(
```

Comment:
Can this be achieve by `ModalBottomSheetRoute.isDismissible` and `ModalBottomSheetRoute.barrierLabel`? This seems to do the same this as the modal barrier.